api/document/tree.py

```diff
@@ -21,6 +23,18 @@ def __getattr__(self, attr):
         """Delegate fields/methods to wrapped model."""
         return getattr(self.model, attr)
 
+    @property
+    def _prefetched_objects_cache(self):
+        """Django REST Framework thinks we're a bona fide Django model,
+        and its UpdateModelMixin sometimes writes to the model's
+        undocumented _prefetched_objects_cache property, so we'll
+        need to make it writable too."""
```

Comment:
Um, needless to say, this feels _super_ sketchy and I'm not sure if there's a cleaner way to get around it.

Reply:
Hmm, one potential way we could work around this may be by converting `DocCursorSerializer` to inherit from the standard `Serializer` rather than `ModelSerializer`, but somehow using a separate `ModelSerializer` subclass to save just the parts of the `DocNode` model that we need it for. I could try experimenting with this.

Reply:
I think switching to the `Serializer` base class makes sense; we've only been using `ModelSerializer` for simplicity when generating fields. I think this nastiness out-weights that.